modules/core/02-client/keeper/grpc_query.go

```diff
@@ -328,3 +328,55 @@ func (k Keeper) UpgradedConsensusState(c context.Context, req *types.QueryUpgrad
 		UpgradedConsensusState: protoAny,
 	}, nil
 }
+
+// VerifyMembership implements the Query/VerifyMembership gRPC method
+func (k Keeper) VerifyMembership(c context.Context, req *types.QueryVerifyMembershipRequest) (*types.QueryVerifyMembershipResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	if err := host.ClientIdentifierValidator(req.ClientId); err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
```

Comment:
I think we should error if the client type is solo machine?

Reply:
localhost as well?

Reply:
do we also want to error if the client is not active, i.e.

`k.GetClientStatus(cachedCtx, clientState, req.ClientId) != exported.Active`

Reply:
> I think we should error if the client type is solo machine?

This would involve adding solomachine types as a concrete dependency in 02-client which I think is currently not the case, (it is already with localhost though).

Reply:
> I think we should error if the client type is solo machine?

Why would we error if it is solomachine client type? State changes are discarded so no sequence incrementing will happen. Shouldn't it be fine to let it go through, just sig verify on whatever bytes? Is it that solomachine is only signing one particular set of bytes for a one sequence? i.e. you will be able to verify for example, a channel end signed at sequence x? so in theory the `VerifyMembership` query is only available for proof one value at a particular point in time. Does that make sense?

> localhost as well?

Yeah, if you query a chain for a proof and have access to an rpc for a header to get the app hash then you should be able to verify off chain and it would be spammy(?) I guess to allow a query to go through like that. Fee like there's no reason to allow it.

> This would involve adding solomachine types as a concrete dependency in 02-client which I think is currently not the case, (it is already with localhost though)

Solomachine client type is in `exported` afaik.

Reply:
Created this issue: https://github.com/cosmos/ibc-go/issues/5848

Reply:
solomachine client type is a string, it can be duplicated

> Is it that solomachine is only signing one particular set of bytes for a one sequence?

yes it is.